src-self-hosted/link.zig

```diff
@@ -31,11 +29,26 @@ pub const Options = struct {
     /// Used for calculating how much space to reserve for executable program code in case
     /// the binary file deos not already have such a section.
     program_code_size_hint: u64 = 256 * 1024,
+    default_entry_addr: u64 = 0x8000000,
+    /// Tracks usable sections of the address space; this is used to determine
+    /// where to place code and data.
+    address_space: ?std.ArrayList(AddressSpaceEntry) = null,
```

Comment:
I'd like to see a more limited API for starters. Instead of this array list, I suggest something along the lines of:

```zig
/// Executable program data.
text_vaddr_range: ?AddressRange = null,
/// Mutable global variables.
data_vaddr_range: ?AddressRange = null,
/// Global constants
rodata_vaddr_range: ?AddressRange = null,
/// Mutable global variables initialized to zero
bss_vaddr_range: ?AddressRange = null,

pub const AddressRange = struct {
    start: u64,
    end: u64,
};
```

The end-game here would be having some data structure akin to a linker script. This code will be highly volatile as the project gains features bringing us closer to that end-game. And so my suggestion is to do the bare minimum you can here to get SPU Mk II working. What are your actual needs for that? Please explain in the PR description.